fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryComparatorPlan.java

```diff
@@ -0,0 +1,227 @@
+    private static final ObjectPlanHash BASE_HASH = new ObjectPlanHash("Record-Query-Comparator-Plan");
+
+    @Nonnull
+    private final KeyExpression comparisonKey;
+    private final int referencePlanIndex;
+
+    private RecordQueryComparatorPlan(@Nonnull final List<Quantifier.Physical> quantifiers,
+                                      @Nonnull final KeyExpression comparisonKey,
+                                      final int referencePlanIndex) {
+        super(quantifiers);
+        this.comparisonKey = comparisonKey;
+        this.referencePlanIndex = referencePlanIndex;
+    }
+
+    /**
+     * Factory method to create a new instance of the Comparator plan.
+     *
+     * @param children the list of plans to compare results from
+     * @param comparisonKey a key expression by which the results of all plans are compared by
+     * @param referencePlanIndex the index of the "reference plan" (source of truth) among the given sub-plans
+     *
+     * @return a new plan that will compare all results from child plans
+     */
+    @Nonnull
+    public static RecordQueryComparatorPlan from(@Nonnull List<? extends RecordQueryPlan> children,
+                                                 @Nonnull KeyExpression comparisonKey,
+                                                 final int referencePlanIndex) {
+        Verify.verify(!children.isEmpty());
+        Verify.verify((referencePlanIndex >= 0) && (referencePlanIndex < children.size()));
+        final ImmutableList.Builder<ExpressionRef<RecordQueryPlan>> childRefsBuilder = ImmutableList.builder();
+        for (RecordQueryPlan child : children) {
+            childRefsBuilder.add(GroupExpressionRef.of(child));
+        }
+        return new RecordQueryComparatorPlan(Quantifiers.fromPlans(childRefsBuilder.build()), comparisonKey, referencePlanIndex);
+    }
+
+    /**
+     * Execute children and compare results.
+     * @param store record store from which to fetch records
+     * @param context evaluation context containing parameter bindings
+     * @param continuation continuation from a previous execution of this same plan
+     * @param executeProperties limits on execution
+     * @param <M> the type of records in the store
+     * @return {@link RecordCursor} that iterates through the results of the execution
+     */
+
+    @Nonnull
+    @Override
+    @SuppressWarnings("squid:S2095")
+    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
+                                                                     @Nonnull final EvaluationContext context,
+                                                                     @Nullable final byte[] continuation,
+                                                                     @Nonnull final ExecuteProperties executeProperties) {
+        final ExecuteProperties childExecuteProperties = executeProperties.clearSkipAndLimit();
```

Comment:
The intent here is to have the limit on the overarching plan rather than on the children. The child plans should return the same results, but we can't assume that they will all follow the same access patterns (e.g. they may scan different indexes). So the idea was to limit the scan by the parent plan (such that the limit takes all children's scans into account).
Similar pattern was used by `RecordQueryIntersectionPlan` - please let me know if this is not the right way.

Reply:
I think the difference is that for intersection plans, not clearing the skip and limit is semantically incorrect (i.e., if you had a limit and/or skip you wanted to enforce, if you enforced it on each child individually, because you don't know which values will or won't be in the intersection until you've read them from the children, you might have to read more data from each child before the limit and/or skip is actually correctly satisfied).

For this plan, the idea is that each of the candidates should be identical in what they return (regardless of what data they access), in which case enforcing a limit should be accomplishable by enforcing the limit on each child individually. Maybe we still want to enforce the limit on the comparator instead, though, I'm not sure.

Reply:
Agree. Changed the skip and limit to be enforced by the child plan and cleared it from the parent plan - this way we can verify that each child is applying it correctly. Documented the logic.